### PR TITLE
Added new search pattern components

### LINKED ISSUE #441: Add documentation for each Search pattern element into the new 'Search' page, under Patterns in the Design System.
### About

**What does it do?**
* Allows the user to search for information across the West Midlands Network website

**When to use it?**
* When a user needs to find the information they need, which they cannot find through the main navigation

**How it works**
* Users should only be able to access the search function through the header by hovering or selecting the magnifying glass icon
* The search bar will appear in a dropdown along with the ‘I want to…’ feature, where users can access popular search queries
* Once a user submits their search query, they will be able to view the search results. They can sort and filter the search results on the search results page

### Search bar

> Insert search bar pattern

### Sort

**What does it do?**
* This gives the user the option to sort search results by relevance or when a page was last updated

> Insert sort pattern

### Filter

**What does it do?**
* This gives the user the option to filter search results using a predetermined set of filters

**How it works**
* The filters will narrow down the search results by displaying pages which only match the filter criteria
* On mobile, the filter options will appear when a user clicks on the ‘Filter your results button’. This will trigger an overlay where users can select and apply filters. This is to reduce page size on smaller devices

> Insert filter collapsed/expanded pattern

### Search result

**What does it do?**
* This is how a matching page will display in the search results page

**How it works**
* Each page result will show the page breadcrumbs on desktop. On mobile, breadcrumbs are not shown. This is to reduce page size on smaller devices
* The page description will display main body copy which is highest on the page. The description will be a maximum of 150 characters. Any descriptions longer than 150 characters will be truncated.
* If the user’s search query is in the page description, it will be highlighted in bold. This is to help them understand the context of the page
* The horizontal rule at the bottom of a search result is to make the results list easier to scan.

> Insert search result pattern

### Pagination links

**What does it do?**
* Allows the user to navigate through search results.

> Insert pagination patterns

### PR DESCRIPTION
**Pull request by houbly**
Fixes #441 
Includes: 
- Search bar
- Sort
- Filter
- Search result
- Pagination

### CODE REVIEW

**Review by gldgrnt — COMMENTED** [review]
All looks good to me. Just wondering if it's worth using `wmndsIcon()` macro in the .njk files just in case the icons get updated in future.

File: src/wmnds/patterns/search/search-filter/_search-filter.scss [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+
+    &__header-title {
+      @include type-setting(2, 0.85);
+      margin: 0;
+      color: get-color(primary);
+    }
+
+    .wmnds-accordion__summary {
+      .wmnds-accordion__summary-title {
+        font-size: 1.4375rem;
```
Comment (catiarodriguescosta) [inline src/wmnds/patterns/search/search-filter/_search-filter.scss:82]: Can't we round it?

**Review by catiarodriguescosta — CHANGES_REQUESTED** [review]
Didn't have time to dig, but it looks pretty well. 
Just left a tiny suggestion - that could not make sense in this particular case.

File: src/wmnds/patterns/search/search-filter/_search-filter.scss (hunk 1 above)
Comment (houbly) [reply src/wmnds/patterns/search/search-filter/_search-filter.scss:82]: Sure. I have no Idea how that got there

**Review by catiarodriguescosta — APPROVED** [review]
Looks good to me! 👍

**Review by jkmnch — APPROVED** [review]

**Review by gldgrnt — APPROVED** [review]
Looks good!

**Pull request merged**